### ISSUE TITLE
Support Java Interop API used by Graal (and Nashorn)

### ISSUE THREAD

**Issue by p-bakker** [mozilla/rhino#1058]
Both Nashorn and GraalVM support options to make migrating from Rhino easier. 

I'd say we should go the other way around: add at least the their Java Interop api (optionally, so behind a flag), so code authored for Graal/Nashorn can be easily ran on Rhino as well :-)

Nice to have though

**Comment by long76** [comment]
yeah, i can't override Nashorn in my Java 8 project because `ReferenceError: "Java" is not defined. (eval#1) in eval at line number 1 at column number 0` (`var MyClass = Java.type("com.example.MyClass");`)

**Comment by p-bakker** [comment]
@long76 Feel like providing a PR for this?

**Comment by tonygermano** [comment]
@szegedi provided a useful link for the Java Global Object API in https://github.com/mozilla/rhino/discussions/972#discussioncomment-1481759

Does anyone know if it Is possible to pass an alternative java ClassLoader to `Java.type(obj)`? I never used Nashorn.

**Comment by p-bakker** [comment]
You mean if that's possible in Nashorn/Graal? Maybe @szegedi knows, as per there mentioned comment he authored the API 🙂

**Comment by p-bakker** [comment]
While adding Javaa.type(...) to aid migration from Nashorn/GraalJS to Rhino, I now think it's implementation could maybe be a small, thin wrapper around the ESM import statement

**Comment by p-bakker** [comment]
@tonygermano could you maybe elaborate on your 'alternative java ClassLoader' comment/requirement? 

Does the current importPackage/Class API in Rhino offer this option?

**Comment by tonygermano** [comment]
I believe it's implemented here
https://github.com/mozilla/rhino/blob/04d54698afd5495d1312daa869a6192898f99ed7/rhino/src/main/java/org/mozilla/javascript/NativeJavaTopPackage.java#L46

```javascript
$ rlwrap java -jar rhino-1.7.13.jar 
Rhino 1.7.13 2020 09 02
js> var url = java.nio.file.Path.of('commons-lang3-3.9.jar').toUri().toURL()
js> var cl = new java.net.URLClassLoader([url])
js> var myPackages = new Packages(cl)
js> Packages.org.apache.commons.lang3.StringUtils
[JavaPackage org.apache.commons.lang3.StringUtils]
js> myPackages.org.apache.commons.lang3.StringUtils
[JavaClass org.apache.commons.lang3.StringUtils]
```
